PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -229,7 +226,17 @@ void ScintillaEditView::init(HINSTANCE hInst, HWND hPere)
 	execute(SCI_SETMARGINMASKN, _SC_MARGE_FOLDER, SC_MASK_FOLDERS);
 	showMargin(_SC_MARGE_FOLDER, true);
 
-	execute(SCI_SETMARGINMASKN, _SC_MARGE_SYBOLE, (1<<MARK_BOOKMARK) | (1<<MARK_HIDELINESBEGIN) | (1<<MARK_HIDELINESEND) | (1<<MARK_HIDELINESUNDERLINE));
+	execute(SCI_SETMARGINMASKN, _SC_MARGE_SYMBOL, (1 << MARK_BOOKMARK) | (1 << MARK_HIDELINESBEGIN) | (1 << MARK_HIDELINESEND) | (1 << MARK_HIDELINESUNDERLINE));
+
+	execute(SCI_SETMARGINMASKN, _SC_MARGE_CHANGEHISTORY, (1 << SC_MARKNUM_HISTORY_REVERTED_TO_ORIGIN) | (1 << SC_MARKNUM_HISTORY_SAVED) | (1 << SC_MARKNUM_HISTORY_MODIFIED) | (1 << SC_MARKNUM_HISTORY_REVERTED_TO_MODIFIED));

```

Comment:
@donho you need to clear the mask from the margin when hiding the margin otherwise the marks once in the margin are show as background in the text.  The same goes for bookmarks.

1) Set a bookmark
2) Settings=>Preferences=>Margins/Border/Edge=> uncheck "Display bookmark"

watch the line background turn the color of the bookmark.  With Change History, it's not immediately noticeable, but when you open a new doc and start typing, you see the background. 

These lines are probably OK here, but you'll need to add:

```C++
execute(SCI_SETMARGINMASKN, _SC_MARGE_SYMBOL, 0);
execute(SCI_SETMARGINMASKN, _SC_MARGE_CHANGEHISTORY, 0);
```

when disabling / hiding the margins.

Reply:
> We could probably use:
> 
> [`SCI_MARKERDELETEALL`](https://www.scintilla.org/ScintillaDoc.html#SCI_MARKERDELETEALL)
> 
> to remove the markers from the margin before hiding it. Maybe ignore it for the Bookmarks, that's a separate (but similar) issue.

Please do not delete Bookmarks when `Display bookmark` is switched off!
BTW, I know and use and like this "bug" for some time...

Reply:
> do not delete Bookmarks when `Display bookmark` is switched off!

Is that current behavior for you @AndiKiissel ?
It is NOT my experience.

Reply:
Yes, 2. is bookmarked, 4. is active line:
![image](https://user-images.githubusercontent.com/107611105/190709403-47cb4fdd-b8b5-483a-9419-e461cb93f0c3.png)
Visible and not so intensive as default blue marker.
I move between them with `F2`/`Shift+F2`.